### PR TITLE
Introduce an extension to Array supporting JSON serialization

### PR DESCRIPTION
**Pull request by nwoolls**

### CODE REVIEW

File: src/Array+Serializable.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+        
+        if NSJSONSerialization.isValidJSONObject(subArray) {
+            do {
+                let json = try NSJSONSerialization.dataWithJSONObject(subArray, options: (prettyPrinted ? .PrettyPrinted : NSJSONWritingOptions()))
+                return json
+            } catch {
+                //currently swift will not catch NSInvalidArgumentException exception
+                print("ERROR: Unable to serialize json, error: \(error)")
+                NSNotificationCenter.defaultCenter().postNotificationName("CrashlyticsLogNotification", object: self, userInfo: ["string": "unable to serialize json, error: \(error)"])
```
Comment (krzyzanowskim) [inline src/Array+Serializable.swift]: this seems unecessary

Comment (nwoolls) [inline src/Array+Serializable.swift]: Which part? The validation was modeled after https://github.com/Mailcloud/swift-serializer/blob/master/src/Serializable.swift#L64

Comment (krzyzanowskim) [inline src/Array+Serializable.swift]: the crashlytics part ;-)

Comment (nwoolls) [inline src/Array+Serializable.swift]: Agreed - I was just picking up the logging patterns found in the Serializable.swift. I'll remove that part.

**Pull request merged**